src/services/users.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Service } from 'typedi';
+import { compareSync, genSaltSync, hashSync } from 'bcrypt';
+import { getRepository } from 'typeorm';
+import { User } from '@entities/user.entity';
+import { JWTService } from '@services/jwt.service';
+import { Errors } from '@constants/errorMessages';
+
+@Service()
+export class UsersService {
+  constructor(private readonly jwtService: JWTService) {}
+
+  private readonly userRepository = getRepository<User>(User);
+
+  givenCrentials(email: string, password: string): boolean {
+    return !!(email && password);
+  }
+
+  comparePassword(password: string, userPass: string): boolean {
+    return compareSync(password, userPass);
+  }
+
+  generateToken(user: User) {
+    return this.jwtService.createJWT(user);
+  }
+
+  hashPassword(password: string): string {
+    return hashSync(password, genSaltSync());
+  }
+
+  async signUp(user: User) {
+    let newUser: User;
+
+    try {
+      newUser = await this.userRepository.save(user);
+    } catch (error) {
+      throw new Error(Errors.MISSING_PARAMS);
+    }
+
+    return newUser;
+  }
+
+  async signIn(email: string, password: string) {
+    if (!this.givenCrentials(email, password)) {
+      throw new Error('Missing params on body');
+    }
```

Comment:
Seems to be logic associated with the request and not too much with the login. The service doesn't need to know that a request was done, it should just check if the credentials are correct, and raise an error if they are not, then the controller can decide how to render that error, but shouldn't be the service responsibility